src/game/game.service.js

```diff
@@ -169,6 +169,28 @@ angular.module('ngGo.Game.Service', [
       this.setTurn((this.info.game.handicap > 1) ? StoneColor.W : StoneColor.B);
     }
 
+    /**
+     * Validate the move is not in the existed children nodes.
+     * @param x X position.
+     * @param y Y position.
+     * @param color Stone color.
+     * @returns {(null|Number)} If the move exists, return the index of the children; otherwise, return null.
+     */
+    function validateMoveNotInChildrenNode(x, y, color) {
```

Comment:
Probably rename this function to `findMoveIndexInChildNodes` to make it more clear what it does and that it returns an index